steps/s3/session.go

```diff
@@ -110,17 +110,17 @@ func (s *Session) ValidateS3BucketExists(ctx context.Context, bucket string) err
 
 	s3Client := s3.New(s.Client)
 	_, err := s3Client.CreateBucket(cparams)
-	if err != nil {
-		if aerr, ok := err.(awserr.Error); ok {
-			if aerr.Code() == s3.ErrCodeBucketAlreadyExists || aerr.Code() == s3.ErrCodeBucketAlreadyOwnedByYou {
-				logger.LogMessage(fmt.Sprintf("validated the existence of bucket: %s", bucket))
-				return nil
-			}
-		} else {
-			return fmt.Errorf("error validating the existence of bucket: %s", bucket)
+
+	if err == nil {
+		return fmt.Errorf("bucket: '%s' does not exists", bucket)
+	}
+	if aerr, ok := err.(awserr.Error); ok {
```

Comment:
errorMsg := fmt.Sprintf("error validating the existence of bucket: %s", bucket")
var aerr awserr.Error
var ok bool
if aerr, ok = err.(awserr.Error); !ok {
  return fmt.Errorf(errorMsg)
}
if aerr.Code() == s3.ErrCodeBucketAlreadyExists || aerr.Code() == s3.ErrCodeBucketAlreadyOwnedByYou {
			logger.LogMessage(fmt.Sprintf("validated the existence of bucket: %s", bucket))
			return nil
}
return fmt.Errorf(errorMsg)

Reply:
Done